torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -294,10 +136,51 @@ InputsIdLookup::IdLookupReturn InputsIdLookup::lookupId(
 FusionExecutorCache::FusionExecutorCache(std::unique_ptr<Fusion> fusion)
     : fusion_(std::move(fusion)) {}
 
+// Note [ Channels Last support in nvfuser ]
+//
+// Background:
+// To support channels last in nvfuser with optimal performance, we would want
+// to allow dimension collapsing in generated code on channels-last tensors,
+// which greatly simplifies indexing. Current API in codegen only allows
+// dimensional collapsing on neighboring axes. The unfortunate thing is that
+// memory format design in PyTorch is implicitly marked by strides, while the
+// semantics meaning of axes remain unchanged. i.e. A 4d tensor with axes [N, C,
+// H, W] would have the same shape in both format, while contiguous tensor
+// carries strides [C*H*W, H*W, W, 1] and channels-last tensor [H*W*C, 1, W*C,
+// C].
+//
+// Approach:
+// Part_1. To allow axes collapsing for channels-last format in codegen, we can
+// permute input tensor to have axes in decending order by their strides, so
+// they would be viewed as `contiguous` in codegen, hence collapsed to simple
```

Comment:
> we can permute input tensor to have axes in decending order by their strides,

This makes sense. Do we need to make it specific to channels last? This seems like a workaround that could be generally useful for tensors whose axes are not ordered in the descending way.

Reply:
Sounds reasonable as a start. We could make it more generic as needed.